### ISSUE TITLE
refresh() not reflecting changes in query

### ISSUE THREAD

**Issue by misterbauer** [nuxt-modules/sanity#530]
I'm attempting to implement a 'Load More' button in my app, and for the life of me I cannot figure out how to get `useSanityQuery` to see the change to `end.value` upon `refresh()`. 

```
<script setup>
let end = ref(1);
let query = groq`*[_type == "product"]|order(orderRank) [0..${end.value}]`;
let { data: products, refresh } = useSanityQuery(query);

function getMoreProducts() {
  end.value++;
  refresh();// I want this to refresh my products list with the updated value of end in the query.
}
</script>
```

Do I need to somehow make `query` reactive? 

Am I doing something wrong here? Am I way off in my thinking on how to accomplish this?

**Comment by danielroe** [comment]
Your query is a string. Changing the value of `end` doesn't make a difference as the string is already 'fixed' at that point.

**Comment by misterbauer** [comment]
> Your query is a string. Changing the value of `end` doesn't make a difference as the string is already 'fixed' at that point.

That makes sense, thank you.

Can you suggest how I might be able to accomplish this instead? I just want to update the parameters in the query and then `refresh()` to retrieve the updated dataset.

Honestly any pointers at all would be greatly appreciated!

**Comment by danielroe** [comment]
I think we can support this in this module by passing in a ref/computed to the params in `useSanityQuery(query, params)`...

**Comment by danielroe** [comment]
This should now be resolved in @nuxtjs/sanity@1.4.1:

```js
const end = ref(1);
const query = groq`*[_type == "product"]|order(orderRank) [0..$end]`
const { data: products, refresh } = useSanityQuery(query, { end })

function getMoreProducts() {
  // this shoud automatically refresh the query
  end.value++
}
```
